modular/executor/execute_gc.go

```diff
@@ -288,21 +290,33 @@ func (e *ExecuteModular) HandleGCObjectTask(ctx context.Context, task coretask.G
 		}
 
 		var redundancyIndex int32 = -1
-		for rIdx, sspId := range gvg.GetSecondarySpIds() {
-			if spId == sspId {
-				redundancyIndex = int32(rIdx)
-				for segIdx := uint32(0); segIdx < segmentCount; segIdx++ {
-					pieceKey := e.baseApp.PieceOp().ECPieceKey(currentGCObjectID, segIdx, uint32(rIdx), objectVersion)
-					if objectInfo.GetRedundancyType() == storagetypes.REDUNDANCY_REPLICA_TYPE {
-						pieceKey = e.baseApp.PieceOp().SegmentPieceKey(objectInfo.Id.Uint64(), segIdx, objectVersion)
+		if len(gvg.GetSecondarySpIds()) != 0 {
+			for rIdx, sspId := range gvg.GetSecondarySpIds() {
+				if spId == sspId {
+					redundancyIndex = int32(rIdx)
+					for segIdx := uint32(0); segIdx < segmentCount; segIdx++ {
+						pieceKey := e.baseApp.PieceOp().ECPieceKey(currentGCObjectID, segIdx, uint32(rIdx), objectVersion)
+						if objectInfo.GetRedundancyType() == storagetypes.REDUNDANCY_REPLICA_TYPE {
+							pieceKey = e.baseApp.PieceOp().SegmentPieceKey(objectInfo.Id.Uint64(), segIdx, objectVersion)
+						}
+						// ignore this delete api error, TODO: refine gc workflow by enrich metadata index.
+						deleteErr := e.baseApp.PieceStore().DeletePiece(ctx, pieceKey)
```

Comment:
can we we "DeletePiecesByPrefix" in this case as well?

Reply:
Done. Updated both primary sp & secondary sp's delete logic to batch delete pieces with object prefix.